### ISSUE TITLE
swift-symbolgraph-extract crashes when trying to emit the 'Swift' module

### ISSUE THREAD

**Issue by kelvin13** [apple/swift#61531]
we used to be able to generate the symbolgraph for the standard library with:

```
swift symbolgraph-extract \
    -target x86_64-unknown-linux-gnu \
    -output-dir . \
    -module-name Swift
```

which worked fine all through the winter, spring, and summer.

but now it just crashes the toolchain:

```
$ swift symbolgraph-extract -target x86_64-unknown-linux-gnu -output-dir . -module-name Swift
swift-symbolgraph-extract: /home/build-user/llvm-project/llvm/include/llvm/ADT/SmallVector.h:291: llvm::SmallVectorTemplateCommon::reference llvm::SmallVectorTemplateCommon<const swift::ProtocolCompositionType *, void>::back() [T = const swift::ProtocolCompositionType *]: Assertion `!empty()' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: /home/ec2-user/.swiftenv/versions/DEVELOPMENT-SNAPSHOT-2022-10-03-a/usr/bin/swift-symbolgraph-extract -target x86_64-unknown-linux-gnu -output-dir . -module-name Swift
1.      Swift version 5.8-dev (LLVM c3aebe50177cd03, Swift 3f020375e568042)
```

**Comment by kelvin13** [comment]
i tested the nightlies from the past month and found that the most recent toolchain that does not crash is `swift-DEVELOPMENT-SNAPSHOT-2022-09-12-a`. we apparently lost the ability to generate symbolgraphs for the standard library starting with `DEVELOPMENT-SNAPSHOT-2022-09-22-a`.

**Comment by QuietMisdreavus** [comment]
@kelvin13 It looks like there should be nightly toolchains with https://github.com/apple/swift/pull/62825 in them now; can you double-check that this was fixed?

**Comment by kelvin13** [comment]
yes, this is now fixed in `swift-5.8-DEVELOPMENT-SNAPSHOT-2023-01-09-a`. thank you!!
